generator/lib/src/realm_model_info.dart

```diff
@@ -27,11 +27,12 @@ class RealmModelInfo {
   final String modelName;
   final String realmName;
   final List<RealmFieldInfo> fields;
+  final ObjectType baseType;
```

Comment:
The name `baseType` is confusing, it sounds like super class type. I'm not sure but, it could be `documentType` for example. Also the enum `ObjectType` could be renamed.